Closes #6591: LCP/ATF data is malformed when no LCP is found by the beacon

Miraeld opened the pull request:
# Description

Fixes #6591

## Documentation

### User documentation

### Technical documentation

I went on the approach to only encode LCP data if this is an array or an object. 
As our `not found` was a string, it was encoded in JSON so we were getting a `"not found"` in the DB while we were checking if this was `=== 'not found'`. That couldn't work. 

Also, I've added a safeguard to make sure we are processing valid lcp data on the front end controller.

And finally, I've added a test to cover this case.

(I based my branch on [Remy's branch](https://github.com/wp-media/wp-rocket/pull/6595) to avoid conflict.)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue).

## New dependencies

## Risks

# Checklists

## Feature validation

- [x] I validated all the Acceptance Criteria. If possible, provide sreenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style
- [x] I wrote self-explanatory code about what it does.
- [x] I wrote comments to explain why it does it.
- [x] I named variables and functions explicitely.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unecessary complexity.
- [x] I listed the introduced external dependencies explicitely on the PR.
- [x] I validated the repo-specific guidelines from CONTRIBUTING.md.

## Observability
- [x] I handled errors when needed.
- [x] I wrote user-facing messages that are understandable and provide actionable feedbacks.
- [x] I prepared ways to observe the implemented system (logs, data, etc.).

remyperona committed:
add debug value to data

remyperona committed:
update test case

remyperona committed:
use debug value

remyperona committed:
remove unused variable

remyperona committed:
update tests

remyperona committed:
fix tests

remyperona committed:
fix tests

Miraeld committed:
Merge branch 'fix/6574-beacon-console' into fix/6591-lcp-atf-data-malformed-when-not-found

Miraeld committed:
Encode LCP to JSON if correct format

Miraeld committed:
Add bail-out safeguard

Miraeld committed:
Added tests

Miraeld commented:
@MathieuLamiot , you'll find two video showing a test in local + on a server (e2e). 

https://github.com/wp-media/wp-rocket/assets/14993267/d310cecc-7bdc-46dd-ba57-ca3c98021bc8

https://github.com/wp-media/wp-rocket/assets/14993267/330b20ca-9b9a-4969-86ef-0ce9ff90d6d6

Miraeld committed:
Fix linter

MathieuLamiot reviewed (changes requested):
@Miraeld When using this branch, I have a message from the beacon when browsing a page without LCP: 

`{success: false, data: 'error when adding the entry to the database'}`

I don't understand why you bail out and return null from create_object, but don't catch this case before assigning the returned value to $lcp?
As a result, I think we get a null for LCP and the DB complains.

Miraeld committed:
Deletes Bail out, and change the way we manage no lcp

Miraeld commented:
I've pushed a modification: 
- I've deleted the bail out, 
- I prevent going through `create_object()` if there is no `type` in the object `image`. 
- I've also refactored the code to make it clearer and simpler.
